src/librustdoc/html/static/main.js

```diff
@@ -442,6 +442,11 @@ function hideThemeButtonState() {
                     // The element is not visible, we need to make it appear!
                     collapseDocs(collapses[0], "show");
                 }
+                // In case this is a sub-variant, toggle the <details> open.
+                var details = h3.parentNode && h3.parentNode.parentNode;
+                if (details && details.tagName === "DETAILS") {
+                    details.open = true;
+                }
```

Comment:
The `parentNode.parentNode` makes me a little uneasy - it seems very tightly tied to the current HTML layout.

It looks like expandSection is called when we're loading a page with a fragment linking directly to a specific item. I think in that case, we can generically do this: climb the HTML tree, and for any ancestor that is a `<details>` tag, set it to open.

Reply:
Makes sense. I introduced another function for that, and fixed the testcase.